src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +109,49 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    protected boolean isRequestROEUpdate(HttpServletRequest request, String transactionId) {
+        final Map<String, List<String>> privileges = getERICTokenPermissions(request);
+        String companyInScope;
+        if (privileges.containsKey(COMPANY_NUMBER_KEY)) {
+            companyInScope = privileges.get(COMPANY_NUMBER_KEY).get(0);
+            if (StringUtils.isNotEmpty(companyInScope) && companyInScope.startsWith("OE")) {
+                String companyInTransaction = getCompanyInTransaction(request, transactionId);
+                if (StringUtils.isNotBlank(companyInTransaction)) {
+                    return companyInScope.equalsIgnoreCase(companyInTransaction);
+                }
+            }
+        }
+        return false;
+    }
+
+    private Map<String, List<String>> getERICTokenPermissions(HttpServletRequest request) {
+        String tokenPermissionsHeader = request.getHeader(ERIC_AUTHORISED_TOKEN_PERMISSIONS_HEADER);
+        Map<String, List<String>> permissions = new HashMap<>();
+        if (tokenPermissionsHeader != null) {
+            for (String pair : tokenPermissionsHeader.split(" ")) {
+                String[] parts = pair.split("=");
+                permissions.put(parts[0], Arrays.asList(parts[1].split(",")));
+            }
+        }
+        return permissions;
+    }
+
+    private String getCompanyInTransaction(HttpServletRequest request, String transactionId) {
+        String passthroughHeader = request.getHeader(ApiSdkManager.getEricPassthroughTokenHeader());
+        String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
+        var logMap = new HashMap<String, Object>();
+        logMap.put(TRANSACTION_ID_KEY, transactionId);
+        try {
+            final var transaction = transactionService.getTransaction(transactionId, passthroughHeader, reqId);
+            logMap.put(COMPANY_NUMBER_KEY, transaction.getCompanyNumber());
+            ApiLogger.debugContext(reqId, "Transaction successfully retrieved " + transactionId, logMap);
+            return transaction.getCompanyNumber();
+        } catch (Exception e) {
+            ApiLogger.errorContext(reqId, "Error retrieving transaction " + transactionId, e, logMap);
+            return null;
```

Comment:
Returning `null` here is not enough/correct. If this happens something has gone wrong with the submission to the end-point (transaction id is wrong) which needs to be reflected with a HTTP 500 error code (see how the `TransactionInterceptor` handles this). So logic/exception handling needs reworking slightly.